src/ddmd/backend/cgcs.c

```diff
@@ -415,7 +415,8 @@ STATIC void ecom(elem **pe)
             dbg_printf("i: %2d Hhash: %6d Helem: %p\n",
                 i,hcstab[i].Hhash,hcstab[i].Helem);
 #endif
-        if (hash == hcstab[i].Hhash && (ehash = hcstab[i].Helem) != NULL)
+        typeof(hcstab[0]) entry = hcstab[i];
```

Comment:
I think auto is not type-inferred in the c++ dialect used by this code
`error: 'entry' does not name a type`